src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java

```diff
@@ -24,7 +25,7 @@
   private final MongoClientSettings settings;
   private final String database;
 
-  public MongoClientOptionsParser(JsonObject config) {
+  public MongoClientOptionsParser(Vertx vertx, JsonObject config) {
```

Comment:
I'll note that this is a break in a public API, but I suspect this mostly gets used internally.

Reply:
i really tried to do it without it... but all variants would end up in a huge refactoring or a lot of redundant code which vertx-core provides out of the box